src/cli_plugin/install/zip.js

```diff
@@ -47,31 +50,32 @@ export function analyzeArchive(archive) {
           return zipfile.readEntry();
         }
 
-        zipfile.openReadStream(entry, function (err, readable) {
+        zipfile.openReadStream(entry, function (error, readable) {
           const chunks = [];
 
-          if (err) {
-            return reject(err);
+          if (error) {
+            return reject(error);
           }
 
           readable.on('data', (chunk) => chunks.push(chunk));
 
           readable.on('end', function () {
-            const contents = Buffer.concat(chunks).toString();
-            const pkg = JSON.parse(contents);
-
-            plugins.push(
-              Object.assign(pkg, {
-                archivePath: match[1],
-                archive: archive,
-
-                // Plugins must specify their version, and by default that version should match
-                // the version of kibana down to the patch level. If these two versions need
-                // to diverge, they can specify a kibana.version to indicate the version of
-                // kibana the plugin is intended to work with.
-                kibanaVersion: get(pkg, 'kibana.version', pkg.version),
-              })
-            );
+            const manifestJson = Buffer.concat(chunks).toString();
+            const manifest = JSON.parse(manifestJson);
+
+            plugins.push({
+              id: manifest.id,
+              stripPrefix: match[1],
```

Comment:
Why the change from `archivePath` to `stripPrefix`? I think the old name was clearer

Reply:
Mostly because `archivePath` sounds like the location of an archive to me, and as far as I can tell this was used for one thing: stripping a prefix from the paths before writing them to disk. I personally like the new name better.